src/Plutarch/Extra/Ord.hs

```diff
@@ -175,6 +200,243 @@ instance DerivePlutusType (PComparator a) where
 
 -- TODO: Semigroup, Monoid
 
+{- | Similar to 'PList', but ensures its contents remain sorted according to the
+ 'POrd' instance of @a@.
+
+ = Note
+
+ We do /not/ derive 'Generic' for this type, as that would leak its
+ internals.
+
+ = Note on conversions
+
+ As 'PSortedList' is not able to be an instance of 'PListLike', if you want to
+ convert to 'PList' specifically, use 'pupcast' instead.
+
+ @since 3.6.1
+-}
+newtype PSortedList (a :: S -> Type) (s :: S)
+    = PSortedList (Term s (PList a))
+
+-- | @since 3.6.1
+instance PlutusType (PSortedList a) where
+    type PInner (PSortedList a) = PList a
+    pcon' (PSortedList t) = t
+    pmatch' x f = f (PSortedList x)
+
+-- | @since 3.6.1
+instance (PEq a) => PEq (PSortedList a) where
+    x #== y = pto x #== pto y
+
+-- | @since 3.6.1
+instance (PShow a) => PShow (PSortedList a) where
+    pshow' b = pshow' b . pto
+
+-- TODO: PPartialOrd, POrd, Semigroup, Monoid
+
+{- | Constructs an empty 'PSortedList'.
+
+ @since 3.6.1
+-}
+psempty ::
+    forall (a :: S -> Type) (s :: S).
+    Term s (PSortedList a)
+psempty = pcon $ PSortedList pnil
+
+{- | Constructs a singleton 'PSortedList'.
+
+ @since 3.6.1
+-}
+pssingleton ::
+    forall (a :: S -> Type) (s :: S).
+    Term s (a :--> PSortedList a)
+pssingleton = phoistAcyclic $
+    plam $ \x ->
+        pcon . PSortedList $ psingleton # x
+
+{- | Inserts an element into an existing 'PSortedList'.
+
+ @since 3.6.1
+-}
+psinsert ::
+    forall (a :: S -> Type) (s :: S).
+    (POrd a) =>
+    Term s (a :--> PSortedList a :--> PSortedList a)
+psinsert = phoistAcyclic $
+    plam $ \x xs ->
+        phandleList (pto xs) (pssingleton # x) $ \y ys ->
```

Comment:
Need to use `pupcast` to be consistent with our comment that encourages using `pupcast`.

Reply:
Nobody will see this but us.